packages/core/src/components/axes/zoom-bar.ts

```diff
@@ -167,21 +167,27 @@ export class ZoomBar extends Component {
 			// update initialZoomDomain and set zoomDomain in model only if the option is changed
 			// not the same object, and both start date and end date are not equal
 			if (
+				newInitialZoomDomain &&
 				!(
-					oldInitialZoomDomain === newInitialZoomDomain ||
-					(oldInitialZoomDomain &&
-						newInitialZoomDomain &&
-						oldInitialZoomDomain[0].valueOf() ===
-							newInitialZoomDomain[0].valueOf() &&
-						oldInitialZoomDomain[1].valueOf() ===
-							newInitialZoomDomain[1].valueOf())
+					oldInitialZoomDomain &&
+					oldInitialZoomDomain[0].valueOf() ===
+						newInitialZoomDomain[0].valueOf() &&
+					oldInitialZoomDomain[1].valueOf() ===
+						newInitialZoomDomain[1].valueOf()
 				)
 			) {
 				this.model.set(
 					{
-						initialZoomDomain: newInitialZoomDomain,
+						// use a new object instead of newInitialZoomDomain
+						initialZoomDomain: [
+							new Date(newInitialZoomDomain[0].getTime()),
+							new Date(newInitialZoomDomain[1].getTime())
+						],
 						zoomDomain: newInitialZoomDomain
-							? newInitialZoomDomain
+							? [
+									new Date(newInitialZoomDomain[0].getTime()),
+									new Date(newInitialZoomDomain[1].getTime())
+							  ]
```

Comment:
so the changes to the if-statement above makes sense, but what are we achieving here? if we really need a new object should we use `Tools.merge({}, newInitialZoomDomain)`?

Reply:
`Tools.merge` makes sense to me. Updated in https://github.com/carbon-design-system/carbon-charts/pull/911/commits/9c959c413b7a7df918609ecca286fe9f1069d004.